pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -33,21 +33,23 @@ const (
 type CommonPrometheusFields struct {
 	// PodMetadata configures Labels and Annotations which are propagated to the prometheus pods.
 	PodMetadata *EmbeddedObjectMetadata `json:"podMetadata,omitempty"`
-	// ServiceMonitors to be selected for target discovery. *Deprecated:* if
-	// neither this nor podMonitorSelector are specified, configuration is
-	// unmanaged.
+	// ServiceMonitors to be selected for target discovery. *Note:* if
+	// neither this nor podMonitorSelector and probeMonitor are specified,
+	// prometheus-operator will not manage the configuration.
```

Comment:
I'd keep the warning that the unmanaged mode is deprecated and will go away in the future.

```suggestion
	// ServiceMonitors to be selected for target discovery.
	//
	// If `spec.serviceMonitorSelector`,
        // `spec.podMonitorSelector` and `spec.probeSelector` are null, the Prometheus
        // configuration is unmanaged. The Prometheus operator will ensure that the
        // Prometheus configuration's Secret exists but it is the responsibility of
        // the user to provide the raw gzipped Prometheus configuration under the
        // `prometheus.yaml.gz` key. This behavior is deprecated and will be removed in the next major version of the custom resource definition. It is recommended to
        // use `spec.additionalScrapeConfigs` instead.

```

Reply:
Done in the last revision, also copied that comment to the other selectors.